akka-docs/src/main/paradox/additional/rolling-updates.md

```diff
@@ -0,0 +1,93 @@
+
+One approach is carried out in two phases: 
+
+1. Deploy a version of your application that still supports deserializing messages in the old form 
+(your deserializer is still in "akka.actor.serializers"-section of your configuration, but it no longer serializes new messages 
+(it is removed from “akka.actor.serialization-bindings”-section). See @ref:[Serialization Configuration](../serialization.md#configuration) for details.
+1. After this upgrade has succeeded and there are no longer systems sending messages serialized with the decommissioned serializer, 
+you can upgrade to a version that also no longer contains this serializer in `akka.actor.serializers`.
+
+## Cluster Sharding
+
+During a rolling upgrade, sharded entities receiving traffic may be moved during @ref:[shard rebalancing](../cluster-sharding.md#shard-rebalancing), 
+to an old or new node in the cluster, based on the pluggable allocation strategy and settings.
+When an old node is stopped the shards that were running on it may be allocated to one of the
+other old nodes remaining in the cluster. 
+ 
+It is recommended to leave the oldest node until last, because the shard coordinator 
+is running on that node. See @ref:[Cluster Shutdown](#cluster-shutdown) below for more detail.
+
+There are some cases when @ref:[a full cluster restart is needed](../cluster-sharding.md#rolling-upgrades).
+
+## Migrating Untyped to Typed
+
+It is recommended with a two step approach:
+
+* Deploy with the new nodes set to `akka.cluster.configuration-compatibility-check.enforce-on-join = off`
+and ensure all nodes are in this state
+* Deploy again and with the new nodes set to `akka.cluster.configuration-compatibility-check.enforce-on-join = on`. 
+
+The configuration from existing nodes should pass the @ref:[Cluster Configuration Compatibility Checks](#cluster-configuration-compatibility-check).
+Find out more about coexisting and @ref:[untyped to typed](../typed/coexisting.md#untyped-to-typed). 
+
+### With Cluster Sharding and Persistence
+
+Rolling upgrades where shards on old nodes are running untyped persistent actors 
+and new ones are running typed persistent behaviors have been tested successfully by the team and users.
+Samples coming soon.
+
+## Cluster Shutdown
+ 
+@ref:[Coordinated Shutdown](../actors.md#coordinated-shutdown) will automatically run when the cluster node sees itself as Exiting.
+@ref:[Graceful shutdown](../cluster-sharding.md#graceful-shutdown) of Cluster Singletons and Cluster Sharding similarly happen automatically.
+ 
+But in case of network failures during rolling restarts and updates it 
+may still be necessary to set the node’s status to Down in order to complete the removal. 
+
+Find out more about
+* @ref:[Cluster Downing](../cluster-usage.md#downing) and providers
+* [Cluster Bootstrap](https://doc.akka.io/docs/akka-management/current/bootstrap/index.md#rolling-updates) and Rolling updates
+
+### Cluster Shutdown Order
```

Comment:
This is not specific to typed and already mentioned above, so perhaps this subsection is not needed?